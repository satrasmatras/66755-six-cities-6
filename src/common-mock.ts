import Offer from "./models/offer";
import Comment from "./models/comment";
import City from "./models/city";
import SortType from "./models/sort-type";
import Routes from "./routes";
import {AuthorizationStatus, LoginPayload} from "./store/user/slice";
import AuthInfo from "./models/auth-info";
import {adaptDataToOffer} from "./adapters/offers";
import {adaptDataToComment} from "./adapters/comments";
import {CommentPost} from "./models/comment-post";
import {adaptDataToAuthInfo} from "./adapters/auth-info";
import {RootState} from "./store";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {redirectMiddleware} from "./store/redirect";

export const MOCK_OFFER_FROM_API = {
  "bedrooms": 3,
  "city": {
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    },
    "name": `Amsterdam`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "avatar_url": `img/1.png`,
    "id": 3,
    "is_pro": true,
    "name": `Angelina`
  },
  "id": 1,
  "images": [`img/1.png`, `img/2.png`],
  "is_favorite": false,
  "is_premium": false,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  },
  "max_adults": 4,
  "preview_image": `img/1.png`,
  "price": 120,
  "rating": 4.8,
  "title": `Beautiful & luxurious studio at great location`,
  "type": `apartment`
};
export const MOCK_ADAPTED_OFFER: Offer = adaptDataToOffer(MOCK_OFFER_FROM_API);

export const MOCK_OFFERS_FROM_API = [
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API
].map((offer, i) => ({
  ...offer,
  id: i + 1
}));

export const MOCK_ADAPTED_OFFERS: Offer[] = MOCK_OFFERS_FROM_API.map(adaptDataToOffer);

export const MOCK_COMMENT_FROM_API = {
  "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "date": `2019-05-08T14:13:56.569Z`,
  "id": 1,
  "rating": 4,
  "user": {
    "avatar_url": `img/1.png`,
    "id": 4,
    "isPro": false,
    "name": `Max`
  }
};

export const MOCK_COMMENTS_FROM_API = [
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API
].map((comment, i) => ({
  ...comment,
  id: i + 1
}));

export const MOCK_ADAPTED_COMMENT: Comment = adaptDataToComment(MOCK_COMMENT_FROM_API);

export const MOCK_ADAPTED_COMMENTS: Comment[] = MOCK_COMMENTS_FROM_API.map(adaptDataToComment);

export const MOCK_CITY: City = {
  "location": {
    "latitude": 52.370216,
    "longitude": 4.895168,
    "zoom": 10
  },
  "name": `Amsterdam`
};

export const MOCK_SORT_TYPE = SortType.TOP_RATED;

export const MOCK_ROUTE = Routes.FAVORITES;

export const MOCK_AUTH_INFO_FROM_API = {
  "avatar_url": `img/1.png`,
  "email": `Oliver.conner@gmail.com`,
  "id": 1,
  "is_pro": false,
  "name": `Oliver.conner`
};

export const MOCK_ADAPTED_AUTH_INFO: AuthInfo = adaptDataToAuthInfo(MOCK_AUTH_INFO_FROM_API);

export const MOCK_AUTHORIZATION_STATUS = AuthorizationStatus.AUTH;

export const MOCK_EMPTY_ACTION = {
  type: `test`,
  payload: 0
};

export const MOCK_LOGIN_PAYLOAD: LoginPayload = {
  email: MOCK_ADAPTED_AUTH_INFO.email,
  password: `12345`
};

export const MOCK_COMMENT_POST: CommentPost = {
  comment: `Lalalalalalallaallaa`,
  rating: 5,
};

export const MOCK_INITIAL_STATE: RootState = {
  offers: {
    offers: MOCK_ADAPTED_OFFERS,
    isLoading: false,
    sortType: MOCK_SORT_TYPE
  },
  city: {
    city: MOCK_CITY
  },
  map: {
    hoveredOffer: MOCK_ADAPTED_OFFER
  },
  user: {
    authorizationStatus: MOCK_AUTHORIZATION_STATUS,
    authInfo: MOCK_ADAPTED_AUTH_INFO
  },
  offer: {
    offer: MOCK_ADAPTED_OFFER,
    offerIsLoading: false,
    comments: MOCK_ADAPTED_COMMENTS,
    commentsAreLoading: false
  },
  favorites: {
    favorites: MOCK_ADAPTED_OFFERS,
    isLoading: false
  }
};

export const MOCK_STORE = configureStore([thunk, redirectMiddleware])(MOCK_INITIAL_STATE);
