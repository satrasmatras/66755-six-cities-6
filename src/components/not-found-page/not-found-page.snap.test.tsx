import React from "react";
import {render} from "@testing-library/react";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import NotFoundPage from "./not-found-page";
import {MOCK_INITIAL_STATE} from "../../common-mock";

const mockStore = configureStore([thunk]);
const store = mockStore(MOCK_INITIAL_STATE);

it(`Should NotFoundPage render correctly`, () => {
  const {container} = render(
      <Provider store={store}>
        <Router history={browserHistory}>
          <NotFoundPage />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
