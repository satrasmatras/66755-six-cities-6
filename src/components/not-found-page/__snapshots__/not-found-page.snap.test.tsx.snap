// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NotFoundPage render correctly 1`] = `
<div>
  <div
    style="display: none;"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clip-rule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fill-rule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clip-rule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fill-rule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Oliver.conner"
                      src="img/1.png"
                    />
                  </div>
                  <span
                    class="header__user-name user__name"
                  >
                    Oliver.conner
                  </span>
                </a>
              </li>
              <li
                class="header_nav-item"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Logout
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index"
    >
      <h1
        class="visually-hidden"
      >
        Not Found 
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item"
                href="#"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="page-not-found"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2>
              Oops. Page Not Found
            </h2>
            <a
              class="places__found"
              href="/"
            >
              To main page...
            </a>
          </section>
        </div>
      </div>
    </main>
  </div>
</div>
`;
